judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl

```diff
@@ -14,3 +14,250 @@ rule ViewTablePageDefinition
+            t.label = s.getLabelModifier().value.value;
+        }
+        log.debug("TablePrimitiveColumnFilter: " + t.name);
+}
+
+@lazy
+@greedy
+rule RowLinkDerivedColumn
+    transform s: JSL!RowLinkDeclaration
+    to t: UI!ui::Column {
+        t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/RowLinkDerivedColumn");
+        // we keep the name of the link it self because the relation won't be present in the visual elements only in the data model
+        t.name = s.name;
+        t.format = "%s";
+        if (s.getLabelModifier().isDefined()) {
+            t.label = s.getLabelModifier().value.value;
+        }
+        if (s.getIconModifier().isDefined()) {
+            t.icon = s.equivalent("RowLinkDerivedColumnIcon");
+        }
+        t.width = s.getWidth().isDefined() ? s.getWidth().value : "120";
+
+        /*
+        if (s.customImplementation.isDefined()) {
+            t.customImplementation = s.customImplementation;
+        }
+        if (s.sort.isDefined()) {
+            t.sort = s.getUiSort();
+        }
+        if (s.sortPrecedence.isDefined()) {
+            t.sortPrecedence = s.sortPrecedence;
+        }
+        if (s.formatValue.isDefined()) {
+            t.formatValue = s.formatValue;
+        }
+        */
+        t.attributeType = s.equivalent("RowLinkDerivedAttributeType");
+        t.representsRelation = s.equivalent("RelationType");
+
+        log.debug("RowLinkDerivedColumn: " + t.name);
+}
+
+@lazy
+@greedy
+rule RowLinkDerivedColumnIcon
+    transform s: JSL!RowLinkDeclaration
+    to t: UI!ui::Icon {
+        t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/RowLinkDerivedColumnIcon");
+    	t.iconName = s.getIconModifier().value.value;
+    	t.name = s.name + "Icon";
+
+    	log.debug("RowLinkDerivedColumnIcon: " + t.name);
+}
+
+@lazy
+@greedy
+rule RowLinkDerivedColumnFilter
+    transform s: JSL!RowLinkDeclaration
+    to t: UI!ui::Filter {
+        t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/RowLinkDerivedColumnFilter");
+        t.name = s.name + "Filter";
+        t.attributeType = s.equivalent("RowLinkDerivedAttributeType");
+        if (s.getLabelModifier().isDefined()) {
+            t.label = s.getLabelModifier().value.value;
+        }
+        log.debug("RowLinkDerivedColumnFilter: " + t.name);
+}
+
+@lazy
+@greedy
+rule RowLinkDerivedAttributeType
+    transform s: JSL!RowLinkDeclaration
+    to t: UI!ui::data::AttributeType {
+        t.setId("(jsl/" + s.getId() + ")/RowLinkDerivedAttributeType");
+        t.memberType = UI!ui::data::MemberType#DERIVED;
+        t.isReadOnly = true;
+        t.name = "_text_" + s.name;
+        t.isRequired = false;
+        t.isFilterable = true;
+
+        // TODO investigate if it really does not matter which string type we pick
+        t.dataType = actorDeclaration.equivalent("Application").dataTypes.selectOne(d | d.isTypeOf(UI!ui::data::StringType));
+
+        s.getContainerEquivalentClassType().attributes.add(t);
```

Comment:
same concern as before (modifying source object)

Reply:
It doesn't do that.